lib/appsignal/tracer.ex

```diff
@@ -207,6 +207,25 @@ defmodule Appsignal.Tracer do
     :ok
   end
 
+  """
+  Registers a span as the current span for this process.
+
+  This is necessary when you want to instrument asynchronous work.
+
+      parent = Appsignal.Tracer.current_span()
+
+      list
+      |> Task.async_stream(fn item ->
+        Appsignal.Tracer.register_current(parent)
+        # ...
+      end)
+      |> Stream.run()
+  """
+  @doc false
+  def register_current(span) do
+    register(%{span | pid: self()})
+  end
```

Comment:
We could do with a simple test to see if this does what we expect it, and if it continues doing this whenever we change it in the future.